exporter/prometheusremotewriteexporter/factory.go

```diff
@@ -90,3 +95,23 @@ func createDefaultConfig() configmodels.Exporter {
 		},
 	}
 }
+
+func validateLabelConfig(cfg *Config) error {
+	sanitizedLabels := make(map[string]string)
+	for key, value := range cfg.ExternalLabels {
+		if key == "" || value == "" {
+			return fmt.Errorf("prometheus remote write: external labels configuration contains an empty key or value")
+		}
+
+		// Sanitize label keys to meet Prometheus Requirements
+		if len(key) > 2 && key[:2] == "__" {
+			key = "__" + sanitize(key[2:])
+		} else {
+			key = sanitize(key)
+		}
+		sanitizedLabels[key] = value
+	}
+
+	cfg.ExternalLabels = sanitizedLabels
+	return nil
+}
```

Comment:
Either change name of the func `validateAndSanitizeExternalLabels` or split this into 2 functions.